openpype/hosts/nuke/startup/custom_popup.py

```diff
@@ -0,0 +1,174 @@
+import sys
+import contextlib
+import re
+import nuke
+from PySide2 import QtCore, QtWidgets
+
+
+def get_main_window():
+    """Acquire Nuke's main window"""
+    main_window = None
+    if main_window is None:
+
+        top_widgets = QtWidgets.QApplication.topLevelWidgets()
+        name = "Foundry::UI::DockMainWindow"
+        for widget in top_widgets:
+            if (
+                widget.inherits("QMainWindow")
+                and widget.metaObject().className() == name
+            ):
+                main_window = widget
+                break
+    return main_window
+
+
+class CustomScriptDialog(QtWidgets.QDialog):
```

Comment:
Can we rewrite this it the way it uses nuke prefabricated ux or was there any particular reason we are not using them in this script? https://learn.foundry.com/nuke/developers/63/pythondevguide/custom_panels.html

Reply:
We rewrite this with the nukescripts method for sure. At first I think it is possible to use `popup.py` panel in` openpype.widgets` as the base parent class for this case but it seems some of the functions dont expose and the position of the window can't be edited, so I abandoned to use "popup.py"